### PR TITLE
[android] 11915 sqlite exception unittest fix

### PR DESCRIPTION
**Pull request by seaders**
Extra logging and some fixes for the SQLite Exceptions in UnitTests.

### CODE REVIEW

**Comment by seaders** [comment]
@dominicmarmionswrve & @Sergio-Mira if ye could take a look, this should get rid of those SQLite errors.

File: SwrveSDK/src/main/java/com/swrve/sdk/SwrveBase.java [hunk 1]
```diff
@@ -684,7 +685,13 @@ public void onResponse(RESTResponse response) {
                                 }
 
                                 try {
-                                    JSONObject responseJson = new JSONObject(response.responseBody);
+                                    JSONObject responseJson = null;
+                                    try {
+                                        responseJson = new JSONObject(response.responseBody);
+                                    } catch(JsonSyntaxException e) {
```
Comment (dominicmarmionswrve) [inline SwrveSDK/src/main/java/com/swrve/sdk/SwrveBase.java]: :+1: But catch JSONException instead ? as per:
 http://developer.android.com/reference/org/json/JSONObject.html#JSONObject(java.lang.String)

File: SwrveSDK/src/main/java/com/swrve/sdk/SwrveBase.java [hunk 2]
```diff
@@ -684,7 +685,13 @@ public void onResponse(RESTResponse response) {
...
+                                        SwrveLogger.e("SwrveSDK unable to decode \"" + response.responseBody + "\" with JSON.");
```
Comment (dominicmarmionswrve) [inline SwrveSDK/src/main/java/com/swrve/sdk/SwrveBase.java]: Might be useful to log:
"SwrveSDK unable to decode user_resources_and_campaigns JSON : \"" + response.responseBody

File: SwrveSDK/src/main/java/com/swrve/sdk/SwrveWakefulService.java [hunk 3]
```diff
@@ -28,10 +28,10 @@ protected void onHandleIntent(Intent intent) {
             if (eventsExtras != null && eventsExtras.size() > 0) {
                 handleSendEvents(eventsExtras);
             } else {
-                SwrveLogger.e(LOG_TAG, "SwrveWakefulService: Unknown intent received.");
+                SwrveLogger.e(LOG_TAG, "SwrveWakefulService: Unknown intent received (extras: " + intent.getExtras().get(EXTRA_EVENTS) + ").");
```
Comment (dominicmarmionswrve) [inline SwrveSDK/src/main/java/com/swrve/sdk/SwrveWakefulService.java]: :+1: but intent.getExtras() is nullable, check it first
http://developer.android.com/reference/android/content/Intent.html#getExtras()

File: SwrveSDK/src/main/java/com/swrve/sdk/localstorage/SQLiteLocalStorage.java [hunk 4]
```diff
@@ -124,6 +124,7 @@ public void setSecureCacheEntryForUser(String userId, String category, String ra
 
     private void insertOrUpdate(String table, ContentValues values, String whereClause, String[] whereArgs) {
         if (connectionOpen.get()) {
+            SwrveLogger.i("SQLLite", "Insert/Open into " + database + " isDbLockedByCurrentThread: " + database.isDbLockedByCurrentThread() + " isOpen: " + database.isOpen());
```
Comment (dominicmarmionswrve) [inline SwrveSDK/src/main/java/com/swrve/sdk/localstorage/SQLiteLocalStorage.java:127]: Use a LOG_TAG that has swrve in the name?

File: SwrveSDK/src/main/java/com/swrve/sdk/localstorage/SQLiteLocalStorage.java [hunk 5]
```diff
@@ -161,6 +162,9 @@ public String getCacheEntryForUser(String userId, String category) {
     @Override
     public String getSecureCacheEntryForUser(String userId, String category, String uniqueKey) throws SecurityException {
         String cachedContent = getCacheEntryForUser(userId, category);
+        if(cachedContent == null) {
+            return null;
```
Comment (dominicmarmionswrve) [inline SwrveSDK/src/main/java/com/swrve/sdk/localstorage/SQLiteLocalStorage.java:166]: :+1:

File: SwrveSDK/src/main/java/com/swrve/sdk/localstorage/SQLiteLocalStorage.java [hunk 6]
```diff
@@ -260,6 +264,8 @@ public void setMultipleCacheEntries(List<Entry<String, Entry<String, String>>> c
 
     @Override
     public void close() {
+        SwrveLogger.i("SQLLite", "we're closing database isDbLockedByCurrentThread: " + database.isDbLockedByCurrentThread() + " isOpen: " + database.isOpen());
```
Comment (dominicmarmionswrve) [inline SwrveSDK/src/main/java/com/swrve/sdk/localstorage/SQLiteLocalStorage.java:267]: Is this one needed?

File: SwrveSDKCommon/src/main/java/com/swrve/sdk/rest/RESTClient.java [hunk 7]
```diff
@@ -61,7 +62,12 @@ public void get(String endpoint, IRESTResponseListener callback) {
             urlConnection = addMetricsHeader(urlConnection);
 
             long start = System.nanoTime();
-            urlConnection.connect();
+            try {
+                urlConnection.connect();
+            } catch (ConnectException e) {
```
Comment (dominicmarmionswrve) [inline SwrveSDKCommon/src/main/java/com/swrve/sdk/rest/RESTClient.java]: :+1: but catch IOException?
http://developer.android.com/reference/java/net/URLConnection.html#connect()

**Comment by dominicmarmionswrve** [comment]
:rocket:

File: SwrveSDK/src/main/java/com/swrve/sdk/localstorage/SQLiteLocalStorage.java (hunk 4 above)
Comment (seaders) [inline SwrveSDK/src/main/java/com/swrve/sdk/localstorage/SQLiteLocalStorage.java:127]: I see these guys as temp that will be removed after we get a more reasonable console from Jenkins.

File: SwrveSDK/src/main/java/com/swrve/sdk/localstorage/SQLiteLocalStorage.java (hunk 6 above)
Comment (seaders) [inline SwrveSDK/src/main/java/com/swrve/sdk/localstorage/SQLiteLocalStorage.java:267]: Same as above, I see these guys as temp that will be removed after we get a more reasonable console from Jenkins.

**Comment by seaders** [comment]
All changes in, quick once over @dominicmarmionswrve?  Then I'll merge and eagle-eye it during the Jenkins build.

File: SwrveSDK/src/main/java/com/swrve/sdk/localstorage/SQLiteLocalStorage.java (hunk 4 above)
Comment (dominicmarmionswrve) [inline SwrveSDK/src/main/java/com/swrve/sdk/localstorage/SQLiteLocalStorage.java:127]: OK.

**Pull request merged**